messenger/http_transporter.go

```diff
@@ -44,6 +44,21 @@ var (
 	errNotStarted      = errors.New("HTTP transport has not been started")
 	errTerminal        = errors.New("HTTP transport is terminated")
 	errAlreadyRunning  = errors.New("HTTP transport is already running")
+
+	httpTransport, httpClient = func() (tr *http.Transport, cl *http.Client) {
```

Comment:
Because I'm assigning both vars at the same time, client depends on
transport. Given that assignments are guaranteed to be carried out in
left-to-right order (just learned something new) I guess I don't need the
func.

On Thu, Oct 15, 2015 at 8:42 PM, Karl Isenberg <notifications@github.com>
wrote:

> In messenger/http_transporter.go
> <https://github.com/mesos/mesos-go/pull/198#discussion_r42198909>:
>
> > @@ -44,6 +44,21 @@ var (
> >  	errNotStarted      = errors.New("HTTP transport has not been started")
> >  	errTerminal        = errors.New("HTTP transport is terminated")
> >  	errAlreadyRunning  = errors.New("HTTP transport is already running")
> > +
> > +	httpTransport, httpClient = func() (tr *http.Transport, cl *http.Client) {
>
> why is this done in a function instead of just in a var definition?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/mesos-go/pull/198/files#r42198909>.
>